README.Rmd

```diff
@@ -15,75 +17,67 @@ knitr::opts_chunk$set(
 
-### R side
+The recommended way to install this package is from the R-multiverse community repository:
 
-1. The structure of classes.
-   In py-polars, the strategy is that classes defined on the Rust side (e.g., `PyDataFrame`) are wrapped by classes defined on the Python side (e.g., `DataFrame`).
-   In r-polars, a complex strategy is adopted to update classes created by Rust side/extendr (e.g., `RPolarsDataFrame`) with a lot of custom code.
-   (This is also related to the fact that extendr makes associated functions of Rust structs members of R classes. savvy does not mix associated functions and methods.)
-2. S3 methods first.
-   This is also related to the Rust side, in the current r-polars, generic functions like `as_polars_series` were added later,
-   so there are several places where type conversion from R to Polars is done on the Rust side, making it difficult to understand where the type conversion is done.
-   If type conversion from R to Polars is done with two generic functions, `as_polars_series` and `as_polars_expr`, the code will be much simpler and customization from the R side will be possible.
-3. Error handling.
-   Currently, r-polars has its own Result type on the R side, and error handling is done through it.
-   The backtrace generated that is quite easy to understand, but it is not necessarily easy to use when using polars internally in other packages, such as `testthat::expect_error()`.
-4. Based on `rlang`.
-   Currently, r-polars has no R package dependencies. This is great,
-   but that includes [a degraded copy of `list2()`](https://github.com/pola-rs/r-polars/blob/6eac27a0766d2b6ca92a72c1c7fa76eaeb58bb98/R/dotdotdot.R#L1-L20)
-   instead of the convenient functions in the `rlang` package.
-   `rlang` is a lightweight R package, and I feel that it is more beneficial to depend on the convenient functions of `rlang` than to stick to no dependencies.
-
-1 and 3 are also related to the fact that it is built with extendr, and it seems that switching to savvy is appropriate here as well.
-If we abandon the current Result type on the R side, it is natural to use `rlang` for error handling, so from that perspective, it is reasonable to depend on `rlang` in 4.
+```r
+Sys.setenv(NOT_CRAN = "true")
+install.packages("polars", repos = "https://community.r-multiverse.org")
+```
 
-### Current Status
+## Usage
 
-The directory structure on the Rust side is a complete copy of py-polars.
-The structure of R classes is also the same as py-polars.
+To avoid conflicts with other packages and base R function names, `{polars}`'s
+top level functions are hosted in the `pl` environment, and accessible via the
+`pl$` prefix.
+This means that `polars` queries written in Python and in R are very similar.
 
-The basic classes such as `DataFrame`, `Series`, `Expr`, and `LazyFrame` have been implemented,
-and some functions work correctly.
+For example, rewriting the Python example from <https://github.com/pola-rs/polars> in R:
```

Comment:
```suggestion
For example, rewriting the Python example from <https://github.com/pola-rs/polars#python> in R:
```

So that this goes straight to the example instead of the main page of the repo.